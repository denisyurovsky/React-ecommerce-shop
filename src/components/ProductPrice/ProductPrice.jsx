import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

import { formatPrice } from '../../helpers/utils/formatData';

import styles from './ProductPrice.module.scss';

export const ProductPrice = ({ discountPrice, price }) => {
  return (
    <>
      {discountPrice ? (
        <>
          <Typography
            variant="h5"
            component="p"
            className={styles.discountPrice}
          >
            {formatPrice(discountPrice)}
          </Typography>
          <Typography
            variant="h5"
            component="p"
            className={styles.crossedPrice}
          >
            {formatPrice(price)}
          </Typography>
        </>
      ) : (
        <Typography variant="h5" component="p">
          {formatPrice(price)}
        </Typography>
      )}
    </>
  );
};

ProductPrice.propTypes = {
  discountPrice: PropTypes.number,
  price: PropTypes.number,
};
