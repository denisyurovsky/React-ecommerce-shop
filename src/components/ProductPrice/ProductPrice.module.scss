@import "../../assets/styles/colors.scss";

.crossedPrice {
  margin: 0 !important;
  color: $grey700 !important;
  font-size: 16px !important;
  text-decoration: line-through;
}

@media all and (max-width: 750px) {
  .crossedPrice {
    font-size: 16px !important;
  }
}
