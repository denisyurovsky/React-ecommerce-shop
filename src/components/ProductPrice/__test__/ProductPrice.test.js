import { render } from '@testing-library/react';
import React from 'react';

import { ProductPrice } from '../ProductPrice';

describe('ProductPrice component', () => {
  describe('snapshots', () => {
    it('renders a snapshot with two prices', () => {
      const { asFragment } = render(
        <ProductPrice price={100} discountPrice={10} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
