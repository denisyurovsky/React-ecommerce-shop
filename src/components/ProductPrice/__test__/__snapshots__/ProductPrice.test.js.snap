// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPrice component snapshots renders a snapshot with two prices 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-h5 discountPrice css-ag7rrr-MuiTypography-root"
  >
    10 $
  </p>
  <p
    class="MuiTypography-root MuiTypography-h5 crossedPrice css-ag7rrr-MuiTypography-root"
  >
    100 $
  </p>
</DocumentFragment>
`;
